mitmproxy/contrib/tnetstring.py

```diff
@@ -158,6 +158,8 @@ def _rdumpq(q, size, value):
         write(b':')
         write(span)
         return size + 2 + len(span) + lvalue
+    elif isinstance(value, str):
+        return _rdumpq(q, size, value.encode())
```

Comment:
`flow.get_state`. I printed the state dict on this line. This is how it looks (with a weird mix of strings and bytes):

```

{
    "id": "f559982b-c57f-4093-a0d6-def70c111154", 
    "response": {
        "reason": b"OK", 
        "timestamp_start": 1467552427.8577867, 
        "content": "", 
        "status_code": 200, 
        "headers": (
            (b"header-response", b"svalue"), 
            (b"content-length", b"7")
        ), 
        "timestamp_end": 1467552427.8577871, 
        "http_version": b"HTTP/1.1"
    }, 
    "intercepted": False, 
    "request": {
        "stickycookie": False, 
        "host": b"address", 
        "first_line_format": "relative", 
        "port": 22, 
        "is_replay": False, 
        "stickyauth": False, 
        "http_version": b"HTTP/1.1", 
        "path": b"/path", 
        "timestamp_start": None, 
        "scheme": b"http", 
        "method": b"GET", 
        "headers": (
            (b"header", b"qvalue"), 
            (b"content-length", b"7")
        ), 
        "timestamp_end": None, 
        "content": ""
    }, 
    "type": "http", 
    "error": None, 
    "server_conn": {
        "cert": None, 
        "source_address": {
            "address": ("address", 22), 
            "use_ipv6": True
        }, 
        "timestamp_start": 1.0, 
        "timestamp_ssl_setup": 3.0, 
        "sni": "address", 
        "ip_address": None, 
        "ssl_established": False, 
        "timestamp_end": 4.0, 
        "via": None, 
        "timestamp_tcp_setup": 2.0, 
        "address": {
            "address": ("address", 22), 
            "use_ipv6": True
        }
    }, 
    "client_conn": {
        "timestamp_start": 1.0, 
        "timestamp_ssl_setup": 2.0, 
        "ssl_established": False, 
        "timestamp_end": 3.0, 
        "clientcert": None, 
        "address": {
            "address": ("address", 22), 
            "use_ipv6": True
        }
    }, 
    b"version": (0, 18)
}
```